.github/workflows/promote-to-production.yml

```
@@ -3,40 +3,96 @@ name: Promote to Production
+            echo "Error: Could not read version from package.json"
+            exit 1
+          fi
+          echo "App Version from package.json: $APP_VERSION"
+          echo "app_version=$APP_VERSION" >> $GITHUB_OUTPUT
 
-          # Checkout production branch from remote
-          git fetch origin production
-          git checkout -B production origin/production
+      - name: Update iOS Production Build Number in versions.json
+        id: update_versions_json
+        run: |
```

actionlint (1.7.7)

workflow command "set-output" was deprecated. use `echo "{name}={value}" >> $GITHUB_OUTPUT` instead: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions

(deprecated-commands)